compiler/rustc_ast/src/ast.rs

```diff
@@ -801,8 +801,8 @@ pub enum PatKind {
     /// A reference pattern (e.g., `&mut (a, b)`).
     Ref(P<Pat>, Mutability),
 
-    /// A literal.
-    Lit(P<Expr>),
+    /// A literal, const block or path.
+    Expr(P<Expr>),
```

Comment:
@oli-obk so the `P<Expr>` doesn't actually contain const blocks or paths just yet. You only renamed the variant?

Reply:
No, it always contained these. I'm working towards making that explicit. I wasn't going to touch the ast immediately, but considering that non-lit variants slipped into rustfmt without notice, I should probably do it sooner.

Until then we can just make rustfmt match exhaustively on lit/neg(lit)/constblock/path and panic in the other cases.

Reply:
I had no idea that was the case. Well, thanks for clearing that up.